odoo/osv/expression.py

```diff
@@ -280,16 +280,20 @@ def combine(operator, unit, zero, domains):
     """
     result = []
     count = 0
-    if domains == [unit]:
-        return unit
     for domain in domains:
+        if isinstance(domain, (list, tuple)):
+            domain = normalize_domain(domain)
+        else:
+            # not a domain, skip it
+            # sometimes a None or False is passed like in name_search()
+            assert not domain
+            continue
```

Comment:
This was an error, and should remain an error IMHO.

Reply:
Hello,

I have fixed the other comments, however in the current implementation passing `[None]` does not end in an error because, on line 290, the domain is normalized only when it is not falsy.
I've added a comment, because in name_search args=None by default (I was starting the investigation in v16 and since then the argument is domain=None).

Do you still want me remove that `else` branch and always use normalize_domain?

Best regards!

Reply:
IMO `None` is not a valid domain, period.  I think that in case a domain defaults on `None`, you should use `args or []` when passing that domain to `AND` or `OR`.  Leaving stuff implicit (like ignoring domains like `None`) is almost always a source of silent bugs, which are harder to detect and fix than "crashing" code.

Reply:
I agree, changed :)